src/apps/backend/modules/task/internal/task-reader.ts

```diff
@@ -23,22 +25,59 @@ export default class TaskReader {
     return TaskUtil.convertTaskDBToTask(taskDb);
   }
 
-  public static async getTasksForAccount(params: GetAllTaskParams): Promise<Task []> {
+   public static async getTasksForAccount(
+    params: GetAllTaskParams,
+  ): Promise<Task[]> {
+    const query: any = {
+      account: params.accountId,
+      active: true,
+    };
+
+    if (params.sharedTask !== undefined) {
+      query.sharedTask = params.sharedTask;
+    }
+
+    const totalTasksCount = await TaskRepository.countDocuments(query);
+
+    const paginationParams: PaginationParams = {
+      page: params.page ? params.page : 1,
+      size: params.size ? params.size : totalTasksCount,
+    };
+    const startIndex = (paginationParams.page - 1) * paginationParams.size;
+
+    const tasksDb = await TaskRepository.find(query)
+      .limit(paginationParams.size)
+      .skip(startIndex);
+
+    return tasksDb.map((taskDb) => TaskUtil.convertTaskDBToTask(taskDb));
+  }
+
+
+
+  public static async getSharedTasksForAccount(params: GetAllTaskParams): Promise<Task []> {
+    const sharedTasks = await ShareTaskRequestRepository.find({
+      account: params.accountId,
+      active: true,
```

Comment:
Add this to the query `status: Approved`. 
You can't use other module's repositories here. Instead, you can create a method in the `SharedTaskRequestReader` to get all approved sharedTaskRequests by accountId.